Source/Laser/LaserProfiles.H

```diff
@@ -0,0 +1,198 @@
+#ifndef WARPX_LaserProfiles_H_
+#define WARPX_LaserProfiles_H_
+
+#include <AMReX_REAL.H>
+#include <WarpXParser.H>
+#include <AMReX_ParmParse.H>
+#include <AMReX_Vector.H>
+
+#include <WarpXParser.H>
+
+#include <map>
+#include <string>
+#include <memory>
+#include <functional>
+#include <limits>
+
+namespace WarpXLaserProfiles {
+
+/** Common laser profile parameters
+ *
+ * Parameters for each laser profile as shared among all laser profile classes.
+ */
+struct CommonLaserParameters
+{
+    amrex::Real wavelength; //! central wavelength
+    amrex::Real e_max;  //! maximum electric field at peak
+    amrex::Real theta_stc; //! Angle between polarization (p_X) and direction of spatiotemporal coupling (stc_direction)
```

Comment:
`stc` in `theta_stc` stands for Spatio-Temporal Couplings (e.g., pulse from tilt, created by a combination of temporal chirp and spatial chirp). Since the other parameters (spatial chirp `zeta`, angular chirp `beta` and temporal chirp `phi2`) are in `GaussianLaserProfile`, wouldn't it be more consistent to have `theta_stc` with the others?

Reply:
Actually `stc_direction` could also be in `GaussianLaserProfile`, and the `pp.query` as well as all
```c++
    s = 1.0_rt / std::sqrt(stc_direction[0]*stc_direction[0] + stc_direction[1]*stc_direction[1] + stc_direction[2]*stc_direction[2]);
    stc_direction = { stc_direction[0]*s, stc_direction[1]*s, stc_direction[2]*s };
    Real const dp2 = std::inner_product(nvec.begin(), nvec.end(), stc_direction.begin(), 0.0);
    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(std::abs(dp2) < 1.0e-14,
        "stc_direction is not perpendicular to the laser plane vector");

    // Get angle between p_X and stc_direction                                                                                                                                    
    // in 2d, stcs are in the simulation plane                                                                                                                                    
#if AMREX_SPACEDIM == 3
    theta_stc = acos(stc_direction[0]*p_X[0] +
                     stc_direction[1]*p_X[1] +
                     stc_direction[2]*p_X[2]);
#else
    theta_stc = 0.;
#endif
```
could go there too.

Reply:
Yes, I totally agree!

Now,  `theta_stc` and `stc_direction` are private parameters of the `GaussianLaserProfile`. However, I had to add `amrex::Vector<amrex::Real> p_X` and `amrex::Vector<amrex::Real> nvec` to the common `CommonLaserParameters` struct.

Reply:
Sure, and adding these to `CommonLaserParameters` makes a lot of sense. Thanks!